File: SparkSQL/Actual-Project/SparkEcomLogAnalysis/spark-project/src/main/scala/spark/ActualProject/LogJob/MySQLUtils.scala
package spark.ActualProject.LogJob

import java.sql.{Connection, DriverManager, PreparedStatement}

/**
 * MySQL操作工具类
 */
object MySQLUtils {

  /**
   * 获取数据库的连接
   */
  def getConnection() = {
    DriverManager.getConnection("jdbc:mysql://localhost:3306/spark_stat_project?user=root&password=123456&ssl=false")
  }

  /**
   * 释放数据库资源
   * @param connection
   * @param pstmt
   */
  def release(connection: Connection, pstmt: PreparedStatement) = {
    try {
      if (pstmt != null) {
        pstmt.close()
      }
    } catch {
      case e: Exception => e.printStackTrace()
    } finally {
      if (pstmt != null) {
        pstmt.close()
      }
    }
  }


  def main(args: Array[String]): Unit = {
    println(getConnection())
  }

}

File: SparkSQL/Actual-Project/SparkEcomLogAnalysis/web/src/main/java/com/spark/Utils/MySQLUtils.java
package com.spark.Utils;

import java.sql.*;

/**
 *  操作MySQL的工具类
 */
public class MySQLUtils {

    private static final String USERNAME = "root";
    private static final String PASSWORD = "123456";
    private static final String DRIVERCLASS = "com.mysql.jdbc.Driver";
    private static final String URL = "jdbc:mysql://localhost:3306/spark_stat_project";

    /**
     * 获取数据库的连接
     * @return
     */
    public static Connection getConnection(){
        Connection connection = null;
        try{
            Class.forName(DRIVERCLASS);
            connection = DriverManager.getConnection(URL,USERNAME,PASSWORD);
        }catch (Exception e){
            e.printStackTrace();
        }

        return connection;
    }


    /**
     * 释放资源
     */
    public static void release(Connection connection, PreparedStatement pstmt, ResultSet rs){

        if(rs!=null){
            try {
                rs.close();
            }catch (SQLException e){
                e.printStackTrace();
            }
        }

        if(pstmt!=null){
            try {
                pstmt.close();
            }catch (SQLException e){
                e.printStackTrace();
            }
        }

        if(connection!=null){
            try {
                connection.close();
            }catch (SQLException e){
                e.printStackTrace();
            }
        }


    }


    /**
     *
     */
    public static void main(String[] args) {
        System.out.println(getConnection());
    }

}
